src/wp-includes/media.php

```diff
@@ -5455,6 +5459,23 @@ function wp_get_loading_attr_default( $context ) {
 		return false;
 	}
 
+	// Conditionally skip lazy-loading on images before the loop.
+	if ( 'wp_get_attachment_image' === $context || 'the_post_thumbnail' === $context ) {
+		// Only apply for main query but before the loop.
+		if ( $wp_query->before_loop && is_main_query() ) {
+			// Any image before header ends should not be lazy load.
+			$is_header_loaded = did_action( 'get_header' ) && ! doing_action( 'get_header' );
```

Comment:
`did_action( 'get_header' )` will return `true` as soon as that action is fired, so we don't need the `! doing_action( 'get_header' )` check.